botfront/server/migrations.js

```diff
@@ -400,6 +404,89 @@ Migrations.add({
+    const title = priorPath ? `${priorPath}__${i}` : _id;
+    return [
+        ...acc,
+        `\n## ${title}\n${story || ''}`,
+        ...flattenStoriesForConversion(branches || [], title),
+    ];
+}, []);
+
+const generateStoryUpdates = (stories) => {
+    const storyUpdates = {};
+    stories.forEach(({ story: title, steps }) => {
+        const [_id, ...indices] = title.split('__');
+        const { $set, $unset } = storyUpdates[_id] || { $set: {}, $unset: {} };
+        const pathPrefix = indices.length
+            ? `branches.${indices.join('.branches.')}.`
+            : '';
+        storyUpdates[_id] = {
+            $set: {
+                ...$set,
+                type: 'story',
+                [`${pathPrefix}steps`]: steps,
+            },
+            $unset: {
+                ...$unset,
+                [`${pathPrefix}story`]: '',
+            },
+        };
+    });
+    return storyUpdates;
+};
+
+Migrations.add({
+    version: 12,
+    up: async () => {
+        // check whether BF was never init or migration has already completed
+        // (story key is deleted at last step, which differentiates new and legacy stories)
+        if (!Stories.find({ story: { $exists: true } }).count()) return;
+        const instances = Instances.find({
+            projectId: { $not: { $regex: 'chitchat' } },
+        }).fetch();
+        const host = instances.length === 1
+            ? instances[0].host
+            : process.env.RASA_MIGRATION_INSTANCE_URL;
+        if (!host) {
+            throw new Error(
+                'Could not find Rasa instance to convert stories to Rasa 2.0 format. Please set env var RASA_MIGRATION_INSTANCE_URL.',
+            );
+        }
+        try {
+            const stories = Stories.find(
+                { story: { $exists: true } },
+                { story: 1, _id: 1, branches: 1 },
+            ).fetch();
+            const storiesFlattened = flattenStoriesForConversion(stories).join('\n');
+            const axiosClient = axios.create();
+            const { data: { data = '' } = {} } = await axiosClient.post(
+                `${host}/data/convert/core`,
+                {
+                    data: storiesFlattened,
+                    input_format: 'md',
+                    output_format: 'yaml',
+                },
+            );
+            const { stories: parsedStories } = safeLoad(data);
+            const storyUpdates = generateStoryUpdates(parsedStories);
+            Object.keys(storyUpdates).forEach(_id => Stories.update({ _id }, storyUpdates[_id]));
+            // rebuild indices
+            const allStories = Stories.find().fetch();
+            allStories.forEach((story) => {
+                const { textIndex, events } = indexStory(story);
+                Stories.update({ _id: story._id }, { $set: { type: story.type, textIndex, events } });
+            });
+        } catch (e) {
+            Migrations._collection._collection.update(
```

Comment:
When you run the migration, it only converts stories that were not converted. It's needed because if there's no Rasa instance available when migration tries to run, migration cannot do anything about it.